### PR TITLE
Style form component and render CuratedList

### LINKED ISSUE #41: Fix remaining issues with new curated list form
- [x] Re-fetch list of curated lists after submitting
- [x] Clear fields after submitting
- [x] Resolve lingering issue with post error when changing between fields
- [x] Make form appear when clicking [+] button
- [x] Style component

### PR DESCRIPTION
**Pull request by nwgambee**
## Problem 
this PR addresses the various issues with the add new list form. These include: styling, displaying the form upon button click, and form validation. This PR also formats a few backend files to get the pack file for CuratedList hooked up. 
## Solutions 
Fixed typos to get CuratedList pack to fire, added styling to the add new list form, and added some conditional rendering for the form itself upon button click. 
## Issues Resolved 
#41

### CODE REVIEW

File: app/assets/stylesheets/curated-list-items.scss [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+.show-form-btn_false, {
+  background-color: $bold-blue;
+  color: #fff;
+  cursor: pointer;
+  font-family: $helvetica;
+  font-weight: 900;
+  font-size: .8em;
+  height: 2.2em;
+  width: 8em;
+  border-radius: 10%;
```
Comment (Tristan-Holtz) [inline app/assets/stylesheets/curated-list-items.scss:32]: This is the button appearance that matches the rest of the site's behavior

File: app/javascript/readingList/CuratedList.jsx [hunk 2]
```diff
@@ -1,166 +1,29 @@
-import { h, Component } from 'preact';
```
Comment (Tristan-Holtz) [inline app/javascript/readingList/CuratedList.jsx:1]: Nice refactor

**Review by Tristan-Holtz — APPROVED** [review]

File: app/javascript/packs/CuratedList.jsx [hunk 3]
```diff
@@ -3,15 +3,15 @@ import { getUserDataAndCsrfToken } from '../chat/util';
...
 function loadElement() {
-  console.log('curated list pack')
   getUserDataAndCsrfToken().then(({ currentUser }) => {
-    const root = document.getElementById('reading-list');
+    const root = document.getElementById('curated-list');
+    const curatedListData = JSON.parse(root.dataset.curatedLists);
+    console.log('line 9:', curatedListData[0])
     if (root) {
-      console.log('rendering CuratedList')
+      console.log('rendering CuratedList:', JSON.parse(root.dataset.curatedLists))
```
Comment (DanielEFrampton) [inline app/javascript/packs/CuratedList.jsx:11]: Should we take this out before merging?

File: app/javascript/readingList/CuratedList.jsx [hunk 4]
```diff
@@ -1,166 +1,29 @@
...
-
-    // this.performInitialSearch({
-    //   containerId: 'reading-list',
-    //   indexName: 'SecuredReactions',
-    //   searchOptions: {
-    //     hitsPerPage,
-    //     filters: `status:${statusView}`,
-    //   },
-    // });
+    console.log('props:', this.props)
```
Comment (DanielEFrampton) [inline app/javascript/readingList/CuratedList.jsx:11]: Should we take this out before merging?

File: app/javascript/readingList/readingList.jsx [hunk 5]
```diff
@@ -36,7 +36,7 @@ export class ReadingList extends Component {
   constructor(props) {
     super(props);
     const { availableTags, statusView, curatedLists, username } = this.props;
-    this.state = defaultState({ availableTags, archiving: false, statusView, curatedLists, username });
+    this.state = defaultState({ availableTags, archiving: false, statusView, curatedLists, username, isHidden: true });
```
Comment (DanielEFrampton) [inline app/javascript/readingList/readingList.jsx:39]: What does this change do?

Comment (DanielEFrampton) [reply app/javascript/readingList/readingList.jsx:39]: Ah nevermind I see it below.

File: app/serializers/curated_list_serializer.rb [hunk 6]
```diff
@@ -8,13 +8,10 @@ def to_json(*_args)
...
-        "slug" => curated_list.slug
-        # "articles" => curated_list.articles.map do |article|
-        #   {
-        #     "title" => article.title,
-        #     "slug" => article.slug
-        #   }
-        # end
+        "slug" => curated_list.slug,
+        "articles" => curated_list.articles.map do |article| 
+          article.to_json
```
Comment (DanielEFrampton) [inline app/serializers/curated_list_serializer.rb:13]: Leaving a note here to myself to revisit this, I think there's a better way to include the articles as JS objects without pre-parsing them as JSON.

**Review by DanielEFrampton — APPROVED** [review]
Thanks for tackling this!

**Pull request merged**